src/test/java/com/integrivideo/steps/ProjectSteps.java

```diff
@@ -109,4 +113,22 @@ public void removeDomain(int domainOrderNum) {
     public void clickOnCreateButton() {
         createProjectPage.clickCreateButton();
     }
+
+    @Step
+    public void openProjectUrlByUser(String projectId, String userEmail, String userPassword) {
+        loginSteps.opensUrlAndLogin(Data.PROJECTS_LIST_URL.concat("/").concat(projectId), userEmail, userPassword);
+    }
+
+    @Step
+    public void validatePageNotFoundError(String projectId) {
+        assertThat("Page /app/projects/".concat(projectId).concat(" not found")).isEqualTo(projectNotFoundPage.getErrorDescription());
```

Comment:
It seems that expected and actual are not at the required places

Reply:
Also I don't like the idea of hardcoding string at this level because we can assert some other messages at 404 page.